integration/testrunner/db.go

```diff
@@ -82,34 +99,85 @@ func makeBigQueryClient(ctx context.Context, tb testing.TB, projectID string) *b
 	return client
 }
 
-// This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
-// file. This method assumes that only a single audit log will match, which constitutes success.
-func queryIfAuditLogExistsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) {
+// This calls the database to validate that an audit log exists with expected format and validate how many logs we expect to match, in order to handle streaming use cases.
+// It uses the retries that are specified in the Config file.
+func validateAuditLogsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string, wantNum int, requireJustification bool) {
 	tb.Helper()
+	tb.Logf("querying BigQuery:\n%s", bqQuery.Q)
+	var logEntries []*loggingpb.LogEntry
+	b := retry.NewConstant(cfg.LogRoutingWait)
+	if err := retry.Do(ctx, retry.WithMaxRetries(cfg.MaxDBQueryTries, b), func(ctx context.Context) error {
+		results, err := queryAuditLogs(ctx, tb, bqQuery)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("query error: %w", err))
+		}
+		// Early exit retry if queried log already found.
+		if len(results) == wantNum {
+			logEntries = results
+			return nil
+		}
+		if len(results) > wantNum {
+			tb.Fatalf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+		}
+		tb.Log("Matching entry not found, retrying...")
+		return retry.RetryableError(fmt.Errorf("no matching audit log found in bigquery after timeout for %q", testName))
+	}); err != nil {
+		tb.Fatalf("retry failed: %v.", err)
+	}
+	for i := 0; i < wantNum; i++ {
+		tb.Logf("diffing LogEntry for index %v", i)
+		diffLogEntry(tb, logEntries[i], requireJustification)
+	}
```

Comment:
Nit: I'd prefer just `range logEntries` rather than relying on the `wantNum`.

Reply:
done